www/js/diary/LabelDetailsScreen.tsx

```diff
@@ -62,7 +63,7 @@ const LabelScreenDetails = ({ route, navigation }) => {
               {t('diary.distance')}
             </Text>
             <Text style={{fontSize: 13, fontWeight: 'bold'}}>
-              {`${trip.display_distance} ${trip.display_distance_suffix}`}
+              {`${getFormattedDistance(trip.distance)} ${distanceSuffix}`}
```

Comment:
> I know that in angular, calling functions for formatting is a performance hit because they need to be recomputed on scroll. Is that not the case in ReactJS?

They can be memoized so they are only computed once. I think this would be the most React-friendly way of handling the `display_` properties.
Generally, the React paradigm is that objects should be immutable; it seems that filling in these fields to the trip object would be discouraged and might not play well with the state management architecture. If the precomputation happens immediately after loading and before any interaction, I don't think it will actually cause problems - but still not a very React-congruent approach.

I was thinking about moving away from precomputation and instead having those `display*` fields computed by a memoized function (this would be inside a custom hook that takes the trip/place/ID as parameter).

The secondary benefit of this approach is that the trip object would be more unified in structure with the `composite trip` definition on the server. (and likewise, place <- `confirmed place`).
Soon, I want to actually have CompositeTrip and ConfirmedPlace be explicitly typed so we know exactly what properties they do and do not have. I think this is a bit more achievable if the `display` fields are not tacked onto the same object.

@shankari I would like some advice on whether this is an acceptable approach.

Reply:
sounds good to me. I am generally in favor of functionality first and performance optimization later, but there's no point in writing code which we know is going to be slow because of recomputation overhead.

Reply:
Will do in a later round of rewrites